cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -156,6 +181,61 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 	return errorsutil.NewAggregate(errs)
 }
 
+// allControlPlaneInstancesUpgraded returns true when all the control plane instances have been upgraded completely.
+// Otherwise, it returns false and a list of control palne instances that have not yet been upgraded.
+//
+// NB. This function can only be called after the current control plane instance has been upgraded already.
+// Because it determines whether the other control plane instances have been upgraded by checking whether
+// the kube-apiserver image of other control plane instance is the same as that of this instance.
```

Comment:
so that the addon will got upgraded along with the last control plan instance's upgrade,  for the HA mode, there is a case the cluster is still functionable with partial upgrade of the control plane instances.

Reply:
At this point, addons will run with older versions, as they are kept compatible.